# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: dom/media/ipc/RemoteVideoDecoder.cpp
# Commit: 7ad8396f67cf
# Full Hash: 7ad8396f67cfc38165755438db5cbb78c1e5e5d3
# Author: Matt Woodrow <mwoodrow@mozilla.com>
# Date: 2019-07-01 21:52:17
# Description:
#   Bug 1562395 - Don't try to create sync handles in the RDD process. r=jya
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D36433
# ==============================================================================

diff -r 0b9e5272fe4d -r 7ad8396f67cf dom/media/ipc/RemoteVideoDecoder.cpp
--- a/dom/media/ipc/RemoteVideoDecoder.cpp	Mon Jul 01 09:43:50 2019 +0000
+++ b/dom/media/ipc/RemoteVideoDecoder.cpp	Mon Jul 01 07:58:26 2019 +0000
@@ -54,8 +54,15 @@
       return nullptr;
     }
 
+    // The RDD process will never use hardware decoding since it's
+    // sandboxed, so don't bother trying to create a sync object.
+    TextureFactoryIdentifier ident = aIdentifier;
+    if (XRE_IsRDDProcess()) {
+      ident.mSyncHandle = 0;
+    }
+
     RefPtr<KnowsCompositorVideo> knowsCompositor = new KnowsCompositorVideo();
-    knowsCompositor->IdentifyTextureHost(aIdentifier);
+    knowsCompositor->IdentifyTextureHost(ident);
     return knowsCompositor.forget();
   }
 
@@ -279,11 +286,10 @@
     : RemoteDecoderParent(aParent, aManagerTaskQueue, aDecodeTaskQueue),
       mVideoInfo(aVideoInfo) {
   if (aIdentifier) {
-    // Check to see if we have a direct PVideoBridge connection to the destination
-    // process specified in aIdentifier, and create a KnowsCompositor representing
-    // that connection if so.
-    // If this fails, then we fall back to returning the decoded frames directly
-    // via Output().
+    // Check to see if we have a direct PVideoBridge connection to the
+    // destination process specified in aIdentifier, and create a
+    // KnowsCompositor representing that connection if so. If this fails, then
+    // we fall back to returning the decoded frames directly via Output().
     mKnowsCompositor =
         KnowsCompositorVideo::TryCreateForIdentifier(*aIdentifier);
   }
